src/main/java/com/sombra/edu/springmenteeproject/service/impl/TransferServiceImpl.java

```
@@ -0,0 +1,84 @@
+public class TransferServiceImpl implements TransferService {
+
+    private final TransferRepository transferRepository;
+    private final UserAccountService userAccountService;
+    private final WalletService walletService;
+    private final BalanceService balanceService;
+    private final TransferRequestService transferRequestService;
+
+    @Transactional
+    @Override
+    public Transfer sendMoneyToAnotherUser(MoneyTransferDTO transferDTO) {
+        TransferRequest transferRequest = getTransferRequest(transferDTO);
+        if (!isValidTransfer(transferDTO)) {
+            getTransfer(transferRequest, Status.FAILED);
+            throw new IllegalArgumentException("Invalid transfer request");
+        }
+        balanceService.subtractMoneyByCurrencyAndWalletId(transferDTO.getQuantity(), transferDTO.getSenderWalletId());
+        balanceService.increaseMoneyByCurrencyAndWalletId(transferDTO.getQuantity(), transferDTO.getReceiverWalletId());
+        return getTransfer(transferRequest, Status.SUCCESS);
+    }
+
+    private boolean isValidTransfer(MoneyTransferDTO transferDTO) {
+        if (!userAccountService.checkIsUserAccountExist(transferDTO.getSenderAccountId(), transferDTO.getReceiverAccountId())) {
+            log.error("Account is not exist");
+            return false;
+        }
+        if (!walletService.checkIsWalletsExist(transferDTO.getSenderWalletId(), transferDTO.getReceiverWalletId())) {
+            log.error("Wallet is not exist");
+            return false;
+        }
+        if (!balanceService.isSameCurrencyInWallets(transferDTO.getSenderWalletId(), transferDTO.getReceiverWalletId())) {
+            return false;
+        }
+        BigDecimal senderBalance = balanceService.getUserBalance(transferDTO.getSenderAccountId(), transferDTO.getSenderWalletId());
+        if (senderBalance.compareTo(transferDTO.getQuantity()) < 0) {
+            log.error("The user does not have enough money in the account");
+            return false;
```

Instead of returning false when validation does not pass simply throw an exception with details of what went wrong